bpm/bonita-api/bonita-common-api/src/main/java/org/bonitasoft/engine/bpm/bar/actorMapping/Actor.java

```diff
@@ -10,32 +10,48 @@
  * You should have received a copy of the GNU Lesser General Public License along with this
  * program; if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
  * Floor, Boston, MA 02110-1301, USA.
- **/
-package org.bonitasoft.engine.actor.xml;
+ ******************************************************************************/
+
+package org.bonitasoft.engine.bpm.bar.actorMapping;
 
 import java.util.HashSet;
 import java.util.List;
+import java.util.Objects;
 import java.util.Set;
 
-import org.bonitasoft.engine.bpm.bar.xml.XMLProcessDefinition.BEntry;
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlAttribute;
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlElementWrapper;
+import javax.xml.bind.annotation.adapters.XmlJavaTypeAdapter;
+
+import org.bonitasoft.engine.bar.BEntry;
 
 /**
  * @author Matthieu Chaffotte
  * @author Baptiste Mesta
  */
+@XmlAccessorType(XmlAccessType.FIELD)
 public class Actor {
 
-    private final String name;
-
+    @XmlAttribute
+    private String name;
+    @XmlAttribute
     private String description;
-
+    @XmlElementWrapper(name = "users", required = false)
+    @XmlElement(name = "user")
     private Set<String> users;
-
+    @XmlElementWrapper(name = "groups", required = false)
+    @XmlElement(name = "group")
     private Set<String> groups;
-
+    @XmlElementWrapper(name = "roles", required = false)
+    @XmlElement(name = "role")
     private Set<String> roles;
-
-    private final Set<BEntry<String, String>> memberships;
+    @XmlElementWrapper(name = "memberships", required = false)
+    @XmlJavaTypeAdapter(BEntryAdapter.class)
+    @XmlElement(name = "membership")
+    private Set<BEntry<String, String>> memberships = null;
```

Comment:
Sure I understand that deleting BEntry class is far beyong the purpose of this PR.

But, since you are actually refactoring ActorMapping serialization and deserialization, you may want to use a Set of Membership object that contains group/role properties and get rid of BEntryAdapter here.

I mean, memberships is a Set of memberships. If I read `Set<BEntry<String, String>> memberships` I say myself "What is this Bentry thing", when I read `Set<Membership> memberships` I say myself, "Ok this is a set of memberships"

This is what I would do but maybe there are side effects that I'm not aware of.